### DISCUSSION TITLE
speed estimate coordinate

### DISCUSSION THREAD

**Discussion by Yeodohoon** [roboflow/supervision#1238] [Q&A] [answered]
I am practicing with this example from https://github.com/roboflow/supervision/blob/develop/examples/speed_estimation/ultralytics_example.py. In the example code, there is a section that calculates coordinates C and D with the line SOURCE = np.array([[1252, 787], [2298, 803], [5039, 2159], [-550, 2159]]). I would like to understand in detail how the x-coordinates -550 and 5039 are calculated.

**Comment by LinasKo** [comment]
Hi @Yeodohoon :wave: 

Are you following this guide? https://blog.roboflow.com/estimate-speed-computer-vision/

If not, check it out - it may help clarify the approach.
Let me know if it doesn't help :wink:

**Comment by Yeodohoon** [comment]
Hello @LinasKo 
I watched all the guides and YouTube explanations.
When calculating the coordinates of points C and D in this part of "Math Behind Perspective Transformation", I don't understand how to find the x coordinate.

**Reply by SkalskiP** [reply, marked as answer]
Hi @Yeodohoon 👋🏻 In a case like the one described in the tutorial, where not all points lie on the image, I usually determine the zone geometry by trial and error.

![design-zone-ezgif com-webp-to-png-converter](https://github.com/roboflow/supervision/assets/26109316/a8f75bfc-d01b-4e86-bd75-049dec97e594)

I start by drawing a simple polygon in Colab, such as a 100x100 square. A good idea is to print the image using matplotlib. This displays a scale from which we can estimate the coordinates of the vertices of our polygon.

![Screenshot 2024-05-29 at 13 40 36](https://github.com/roboflow/supervision/assets/26109316/a3fbb804-42be-4ea1-98ce-0d7661138fb9)

Looking at the vertical axis, I conclude that I want to cut off the road around y = 750. From the x-axis, I roughly estimate where points A and B should be. The remaining two points will be on the bottom edge of the image. I know the image has a resolution of (2160, 3840), so for now, I set points C and D at the corners of the image.

![Screenshot 2024-05-29 at 13 47 07](https://github.com/roboflow/supervision/assets/26109316/9d8bda0c-eac5-49a9-ad2e-2ff9a672f0e8)

I wanted points A and B to be exactly on the posts at the roadside. I adjusted their positions with a few attempts.

![Screenshot 2024-05-29 at 13 49 35](https://github.com/roboflow/supervision/assets/26109316/105c3ab4-93dc-4e4e-a162-f215a38241d8)

Now I move points C and D outward so that the lines align with the edge of the road. It takes a few more attempts to get it just right.

![Screenshot 2024-05-29 at 13 52 15](https://github.com/roboflow/supervision/assets/26109316/f9138219-c82b-4f17-8f1d-a0e4d56514c2)

I know it's a rather monotonous and manual method, but it works.
